renderdoccmd/renderdoccmd_linux.cpp

```diff
@@ -195,7 +236,319 @@ void VerifyVulkanLayer(const GlobalEnvironment &env, int argc, char *argv[])
   add_command("vulkanregister", new VulkanRegisterCommand(env));
 }
 
-static Display *display = NULL;
+#if defined(RENDERDOC_WINDOWING_WAYLAND)
+
+static void create_window(uint32_t width, uint32_t height);
+
+static void shm_format(void *data, struct wl_shm *wl_shm, uint32_t format)
+{
+}
+
+static const struct wl_shm_listener shm_listener = {
+  shm_format,
+};
+
+static void pointer_handle_enter(void *data, struct wl_pointer *pointer,
+                                  uint32_t serial, struct wl_surface *surface,
+                                  wl_fixed_t sx, wl_fixed_t sy)
+{
+  struct wl_buffer *buff;
+  struct wl_cursor_image *image;
+
+  image = default_cursor->images[0];
+  buff = wl_cursor_image_get_buffer(image);
+  wl_pointer_set_cursor(pointer, serial, cursor_surface, image->hotspot_x,
+                        image->hotspot_y);
+  wl_surface_attach(cursor_surface, buff, 0, 0);
+  wl_surface_damage(cursor_surface, 0, 0, image->width, image->height);
+  wl_surface_commit(cursor_surface);
+}
+
+static void pointer_handle_leave(void *data, struct wl_pointer *pointer,
+                                  uint32_t serial, struct wl_surface *surface)
+{
+}
+
+static void pointer_handle_motion(void *data, struct wl_pointer *pointer,
+                                  uint32_t time, wl_fixed_t sx, wl_fixed_t sy)
+{
+}
+
+static void pointer_handle_button(void *data, struct wl_pointer *wl_pointer,
+                                  uint32_t serial, uint32_t time,
+                                  uint32_t button, uint32_t state)
+{
+  if (button == BTN_LEFT && state == WL_POINTER_BUTTON_STATE_PRESSED)
+    wl_shell_surface_move(shell_surface, seat, serial);
+}
+
+static void pointer_handle_axis(void *data, struct wl_pointer *wl_pointer,
+                                uint32_t time, uint32_t axis, wl_fixed_t value)
+{
+}
+
+static const struct wl_pointer_listener pointer_listener = {
+  pointer_handle_enter,
+  pointer_handle_leave,
+  pointer_handle_motion,
+  pointer_handle_button,
+  pointer_handle_axis,
+};
+
+static void keyboard_handle_keymap(void *data, struct wl_keyboard *keyboard,
+                                    uint32_t format, int fd, uint32_t size)
+{
+}
+
+static void keyboard_handle_enter(void *data, struct wl_keyboard *keyboard,
+                                  uint32_t serial, struct wl_surface *surface,
+                                  struct wl_array *keys)
+{
+}
+
+static void keyboard_handle_leave(void *data, struct wl_keyboard *keyboard,
+                                  uint32_t serial, struct wl_surface *surface)
+{
+}
+
+static void keyboard_handle_key(void *data, struct wl_keyboard *keyboard,
+                                uint32_t serial, uint32_t time, uint32_t key,
+                                uint32_t state)
+{
+  if (key == KEY_F && state == WL_KEYBOARD_KEY_STATE_PRESSED)
```

Comment:
Please don't add new functionality like this inside a window system implementations. Fullscreen is not necessary or wanted for this, which is only used as a way to quickly sanity check/preview a capture can load without booting the full UI.